### PR TITLE
Lint passing Cow by reference

### LINKED ISSUE #2405: Lint passing `Cow` by reference
See discussion in #2387.

### PR DESCRIPTION
**Pull request by bootandy**
Add lint for reference to Cow to the same place in the code where
the lint for detecting a reference to String lives.

Fixes #2405

### CODE REVIEW

**Comment by bootandy** [comment]
actually the same tests failed without this commit so the current test failures are unrelated

File: clippy_lints/src/ptr.rs [hunk 1]
```diff
@@ -213,6 +213,21 @@ fn check_fn(cx: &LateContext, decl: &FnDecl, fn_id: NodeId, opt_body_id: Option<
                         },
                     );
                 }
+            } else if match_type(cx, ty, &paths::COW) {
+                let as_str = format!("{}", snippet_opt(cx, arg.span).unwrap());
```
Comment (bootandy) [inline clippy_lints/src/ptr.rs]: Is it ok to call unwrap() like this? I could import the function snippet() instead as that doesn't return a Result object

Comment (oli-obk) [reply clippy_lints/src/ptr.rs]: I'll review more next week, but clippy dogfood will disallow the unwrap and the trivial format.

If there is no snippet, we should not be producing a suggestion.

**Comment by clarcharr** [comment]
I'd recommend changing `&Cow<T>` to `&T` personally

File: clippy_lints/src/ptr.rs (hunk 1 above)
Comment (bootandy) [reply clippy_lints/src/ptr.rs]: Interestingly the dogfood test passed for me locally with this code. 

I will update this code as you suggested, thanks.

**Pull request merged**